File: client_angular/src/app/models/commande.ts
import {LigneCommande} from "./ligne-commande";

export interface Commande {
  id?: number;
  utilisateurId: number;
  utilisateurNom?: string;
  lignes: LigneCommande[];
  total: number;
  statut?: string;          // "EN_ATTENTE", "EN_COURS", "LIVREE", etc.
  dateCommande?: string;    // ISO string
}

File: springboot/app/src/main/java/com/bugshan/automative/group/app/model/Commande.java
package com.bugshan.automative.group.app.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Commande {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime dateCommande = LocalDateTime.now();

    private String statut = "EN_ATTENTE";
    // EN_ATTENTE, PAYEE, EN_LIVRAISON, LIVREE, ANNULEE...

    private double total;

    @ManyToOne
    @JoinColumn(name = "utilisateur_id")
    private Utilisateur utilisateur;

    @OneToMany(mappedBy = "commande", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<LigneCommande> lignes = new ArrayList<>();
}
